reboot_toolkit/toolkit_backend.py

```diff
@@ -259,27 +264,54 @@ def list_available_s3_keys(org_id: str, df: pd.DataFrame) -> list[str]:
 
     return all_files
 
-
 def load_games_to_df_from_s3_paths(
-        game_paths: list[str], add_ik_joints: bool = False, add_elbow_var_val: bool = False
+        game_paths: list[str], add_ik_joints: bool = False, add_elbow_var_val: bool = False, game_proportion: float = 1.0
 ) -> pd.DataFrame:
     """
     For a list of paths to the S3 folder of data for each game, load the data into a pandas dataframe.
 
     :param game_paths: list of paths to folders with data for a player
     :param add_ik_joints: whether to add joints necessary to run analyses dependent on IK
     :param add_elbow_var_val: whether to add elbow varus valgus columns set to 0 degrees
+    :param game_proportion: proportion of the player's games to sample from
     :return: dataframe of all data from all games
     """
+
+    movement_sampling_game_threshold = 10
+    movement_sampling_proportion = 0.2
+    movement_sampling_min_movements = 15
+    game_cap = 60
+
     game_paths = sorted(list(set(game_paths)))
 
+    if game_proportion < 1.0:
+        game_sample_size = int(game_proportion * len(game_paths))
+        game_paths = random.sample(game_paths, game_sample_size)
+
+    if len(game_paths) > game_cap:
+        print(f'Number of games selected exceeds request limit. {game_cap} games randomly sampled from date range.')
+        print('  If sample data is not desired, reduce the selection range or contact Reboot for more information.')
+        game_paths = random.sample(game_paths, game_cap)
+
+    if len(game_paths) > movement_sampling_game_threshold:
+        movement_proportion = movement_sampling_proportion
+        min_movements = movement_sampling_min_movements
+    else:
+        movement_proportion = 1.0
+        min_movements = 0
+
     all_games = []
 
     for i, game_path in enumerate(game_paths):
 
         try:
             if 'hitting-processed-series' in game_path:
                 swing_filenames = wr.s3.list_objects(game_path)
+
+                if movement_proportion < 1.0:
+                    movement_sample_size = max(int(movement_proportion * len(swing_filenames)), min(min_movements, len(swing_filenames)))
+                    swing_filenames = random.sample(swing_filenames, movement_sample_size)

```

Comment:
like above, also probably need to sorting for all the file names after sampling

Reply:
Moved sorting to after the game sampling

Reply:
Do we also need to sort the swing and pitch file names after we sample from within the game (as opposed to above where we're sampling across games)?

Reply:
Yeah it looks like this is necessary. Added sorted after game sampling.